app/controllers/reminders_controller.rb

```diff
@@ -122,8 +133,10 @@ def reminder_params
   def one_time_password
     case current_slug
     when "personal-details"
-      ReminderMailer.email_verification(current_reminder, otp.code).deliver_now if current_reminder.valid?(:"email-address")
-      session[:sent_one_time_password_at] = Time.now
+      if current_reminder.valid?(:"personal-details")
```

Comment:
Suggestion: raise exception if invalid and rescue to prevent silent failure.

Reply:
@slorek Yes I noticed that option, but on L31 it will attempt to save with the same context and that will render the inline form error as it can't save, basically it duplicates running the validation with the way the code is laid out.